app/javascript/components/public_health/AdvancedFilter.js

```diff
@@ -0,0 +1,608 @@
+        <Form.Control
+          as="select"
+          value={current}
+          className="py-0 my-0"
+          size="sm"
+          onChange={event => {
+            this.changeFilterDateOption(index, event.target.value);
+          }}>
+          <option value="within">within</option>
+          <option value="before">before</option>
+          <option value="after">after</option>
+        </Form.Control>
+      </Form.Group>
+    );
+  }
+
+  renderFilterNameModal() {
+    return (
+      <Modal
+        show={this.state.showFilterNameModal}
+        centered
+        onHide={() => {
+          this.setState({ showFilterNameModal: false });
+        }}>
+        <Modal.Header>
+          <Modal.Title>Filter Name</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Form.Control
+            as="input"
+            value={this.state.filterName || ''}
+            className="py-0 my-0"
+            onChange={event => {
+              this.setState({ filterName: event.target.value });
+            }}
+          />
+        </Modal.Body>
+        <Modal.Footer>
+          <Button
+            variant="secondary btn-square"
+            onClick={() => {
+              this.setState({ showFilterNameModal: false, show: true, filterName: null });
+            }}>
+            Cancel
+          </Button>
+          <Button
+            variant="primary btn-square"
+            disabled={!this.state.filterName}
+            onClick={() => {
+              this.setState({ showFilterNameModal: false, show: true }, () => {
+                this.save();
+              });
+            }}>
+            Save
+          </Button>
+        </Modal.Footer>
+      </Modal>
+    );
+  }
+
+  // Render a single line "statement"
+  renderStatement(filterOption, value, index, total, dateOption) {
+    return (
+      <React.Fragment key={'rowkey-filter-p' + index}>
+        {index > 0 && index < total && (
+          <Row key={'rowkey-filter-and' + index} className="pb-2 pt-2">
+            <Col className="py-0">
+              <b>AND</b>
+            </Col>
+          </Row>
+        )}
+        <Row key={'rowkey-filter' + index} className="pb-1 pt-1">
+          <Col className="py-0" md="8">
+            {this.renderOptions(filterOption?.name, index)}
+          </Col>
+          {filterOption?.type === 'date' && (
+            <Col className="py-0" md="3">
+              {this.renderDateOptions(dateOption, index)}
+            </Col>
+          )}
+          <Col className="py-0">
+            {filterOption?.type === 'boolean' && (
+              <ButtonGroup toggle>
+                <ToggleButton
+                  type="checkbox"
+                  variant="outline-primary"
+                  checked={value}
+                  value="1"
+                  size="sm"
+                  onChange={() => {
+                    this.changeValue(index, !value);
+                  }}>
+                  TRUE
```

Comment:
thoughts on using "YES/NO" instead of "TRUE/FALSE" for users?

Reply:
Good idea, but I think within the context of the "logic" of this view (particularly the ANDs), I'd prefer to keep it "logic speak" with true and false vs. yes and no. Not a strong argument, but I think that is the case either way.